python/arcticdb/storage_fixtures/azurite_patching.py

```diff
@@ -0,0 +1,79 @@
+import os
+import platform
+import re
+import shutil
+import tempfile
+from pathlib import Path
+from subprocess import run, PIPE
+from typing import List, Optional
+
+_UNIX = platform.system() != "Windows"
+
+# language=js
+_PATCH_JS = """#!/usr/bin/env node
+
+const operation = require("blob/generated/artifacts/operation")
+const perms = require("blob/authentication/OperationBlobSASPermission");
+
+// Should require read permission instead of no permission per
+// https://learn.microsoft.com/en-us/rest/api/storageservices/get-container-properties?tabs=shared-access-signatures#account-sas
+perms.OPERATION_BLOB_SAS_CONTAINER_PERMISSIONS.set(
+    operation.Operation.Container_GetProperties, new perms.OperationBlobSASPermission("r"));
+
+require("blob/main"); // Calls main() on import"""
+
+
+def _write_exec_file(path: Path, content: str):
+    with open(path, "w") as f:
+        f.write(content)
+    if _UNIX:
+        path.chmod(0o755)
+
+
+def _get_azurite_dist_dir(work_dir: Path):
+    # Problem: the `azurite` command could be a shell script and might not be in the node_modules folder
+    # Solution: Use a fake node executable to print out the location of the JS
+    fake_node = work_dir / ("node" if _UNIX else "node.cmd")
+    try:
+        _write_exec_file(fake_node, "#!/bin/sh\necho AZURITE=$1" if _UNIX else "@echo AZURITE=%1\r\n@exit")
+        env = {**os.environ, "PATH": f"{work_dir}{os.pathsep}{os.getenv('PATH')}"}
+        # On Windows, the azurite script may exit with code 1 under our fake node
+        proc = run("azurite", shell=True, env=env, stdout=PIPE, universal_newlines=True, timeout=3, check=_UNIX)
+    finally:
+        fake_node.unlink()
+    match = re.search(r'^AZURITE=("?)(.+)\1$', proc.stdout, flags=re.MULTILINE)
+    assert match, "Running `azurite` with our fake `node` didn't return the expected text: " + proc.stdout
+    resolved = Path(match.group(2).replace("\\\\", "\\")).resolve(strict=True)
+    assert resolved.match("**/node_modules/azurite/dist/src/*"), f"Unusual npm path [{resolved}] from {proc.stdout}"
+    return resolved.parent
+
+
+class PatchedAzurite:
+    """Azurite does not implement some of the documented permission requirements, so we patch it."""
```

Comment:
The problem is that many of the permissions in Azurite are implemented incorrectly, but I only need one fixed for my tests.

If I contribute a patch, then I'll probably need to fix all the similar ones to pass their code review, and it will have wider implications (breaking other user's tests).

Reply:
Can we raise a discussion with them and see if they would be open to the change, at least?

Reply:
Raised https://github.com/Azure/Azurite/issues/2299

I'll remove this code, but it will mean the upcoming PRs will not be able to work until Azurite fixes it.

Reply:
Great thank you. Can we keep this patching mechanism in the git history so we can reinstate it if the Azurite people aren't helpful?

Reply:
You can find all past commits in this PR's time line